service/league_service/typedefs.py

```diff
@@ -26,17 +27,57 @@ class GameOutcome(Enum):
 
 class LeagueDivision(NamedTuple):
     id: int
-    from_mean: float
-    to_mean: float
+    min_rating: float
+    max_rating: float
+    lowest_score: int
     highest_score: int
 
 
+@with_logger
 class League(NamedTuple):
     name: str
     divisions: List[LeagueDivision]
     current_season_id: int
     rating_type: str
 
+    @classmethod
```

Comment:
Maybe to explain, normal methods are like normal methods in other languages (except you have to actually include the `self` in the signature, which is hidden from you in Java or C++ (the `this` pointer in C++)) and `@staticmethod` makes it a static method like in C++ (and probably Java?).
In python however, a class can have state, too (e.g. a variable that is shared between all instances) and if you want access to that state you make it a `@classmethod`.
```python
class League:
  @classmethod
  cls_method(cls):
    pass

  method(self):
    pass

league = League()

# This is the same as calling `League.method` with argument `league`:
league.method()

# These are the same as calling `League.cls_method` with  argument `League`:
League.cls_method()
league.cls_method()
```

This one might be a better explanation:
https://stackoverflow.com/questions/136097/difference-between-staticmethod-and-classmethod/1669524#1669524

Class methods are mostly used for factories, because you want to have access to which class it was called on to instantiate a new object of that class. We used it before simply because `_logger` was an attribute of that class and to have access to the other class methods.